### PR TITLE
Add dependabot config for managing GitHub Action versions

### PR DESCRIPTION
**Pull request by sgibson91**
In the future we may also want to consider applying this to things like environment.yaml files or requirements.txt, etc.

### CODE REVIEW

File: .github/dependabot.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    groups:
+      # Group all GHA updates together into single PR (to avoid desync in related actions e.g. upload-artifact, download-artifact)
+      # ref: https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#groups
+      actions:
+        patterns:
+          - "*"
+    ignore:
+      # Ignore minor upgrades to reduce noise.
+      - update-types: ["version-update:semver-minor"]
+        dependency-name: "*" # Match all packages
```
[the remark this answers is not in the document]
Comment (sgibson91) [reply .github/dependabot.yaml:20]: The way GitHub Actions versioning works is that the `@v3` tag, for example, is kept up-to-date with the minor and patch versions. It's not quite building from the default branch, but will be updated with bugfixes and feature improvements. So we only need to bump every major anyway.

[the remark this answers is not in the document]
Comment (sgibson91) [reply .github/dependabot.yaml:20]: E.g. here v4 and v4.1.7 were both updated on the same day https://github.com/actions/checkout/tags

[the remark this answers is not in the document]
Comment (sgibson91) [reply .github/dependabot.yaml:20]: And dependabot will ignore these bumps anyway if we only use tags like `v4` (which we do)

[the remark this answers is not in the document]
Comment (JimMadge) [reply .github/dependabot.yaml:20]: Ah, good point.

I've found some actions being not so friendly with providing tags like that, but we can worry about that if it becomes a problem :+1:.

**Review by JimMadge — APPROVED** [review]

**Pull request merged**